sys/net/gnrc/link_layer/netdev2/gnrc_netdev2_eth.c

```diff
@@ -109,17 +113,16 @@ static inline void _addr_set_broadcast(uint8_t *dst)
     memset(dst, 0xff, ETHERNET_ADDR_LEN);
 }
 
-#define _IPV6_DST_OFFSET    (36)    /* sizeof(ipv6_hdr_t) - 4  */
-
 static inline void _addr_set_multicast(uint8_t *dst, gnrc_pktsnip_t *payload)
 {
     switch (payload->type) {
-#ifdef MODULE_IPV6
+#ifdef MODULE_GNRC_IPV6
         case GNRC_NETTYPE_IPV6:
+            /* https://tools.ietf.org/html/rfc2464#section-7 */
             dst[0] = 0x33;
             dst[1] = 0x33;
-            memcpy(dst + 2, ((uint8_t *)payload->data) + _IPV6_DST_OFFSET, 4);
-            /* TODO change to proper types when gnrc_ipv6_hdr_t got merged */
+            ipv6_hdr_t *ipv6 = (ipv6_hdr_t*)payload->data;
```

Comment:
From the Linux-kernel conventions we inherit from:

    When declaring pointer data or a function that returns a pointer type, the
    preferred use of '*' is adjacent to the data name or function name and not
    adjacent to the type name.  Examples:
    
    	char *linux_banner;
    	unsigned long long memparse(char *ptr, char **retptr);
    	char *match_strdup(substring_t *s);

Oh, and since you forced me to read the conventions, regarding casting:

    Casting the return value which is a void pointer is redundant. The conversion
    from void pointer to any other pointer type is guaranteed by the C programming
    language.